src/amber/router/segments/variable_segment.cr

```diff
@@ -1,12 +1,19 @@
 module Amber::Router
   class VariableSegment(T) < Segment(T)
-    def initialize(segment, @pattern : Regex? = nil)
+    def initialize(segment, @pattern : (Regex | Symbol)? = nil)
       super segment
     end
 
     def match?(segment : String) : Bool
-      if pattern = @pattern
-        !(segment =~ pattern).nil?
+      case @pattern
+      when :integer
+        match_integer?(segment)
+      when :uuid
+        match_uuid?(segment)
+      when :ascii
+        segment.ascii_only?
```

Comment:
The integer and uuid matching above uses a local private method, but this ascii uses a member method on the segment class. I think I prefer the latter (`segment.ascii_only?`). Is it possible to move `match_uuid?` and `match_integer?` into the segment class?

Reply:
I think this could be a good idea.

No other segment has a match or previously matched against a regex.  Is there a use case for this?

I am treating the segment as a string and there is a builtin method for ascii which is why I used it.